packages/calypso-e2e/src/env-variables.ts

```diff
@@ -33,6 +33,25 @@ const defaultEnvVariables: SupportedEnvVariables = {
 	RETRY_COUNT: 0,
 };
 
+const getAtomicVariationInMixedRun = () => {
+	const allVariations: AtomicVariation[] = [
+		'default',
+		'php-old',
+		'php-new',
+		'wp-beta',
+		'wp-previous',
+		'private',
+		'ecomm-plan',
+	];
+	const currentHour = new Date().getHours();
+	const jestWorkerId = Number( process.env.JEST_WORKER_ID );
+	// We want there to be both some randomness in the variations, but as much of an even distribution as possible.
+	// By combining the current hour plus the Jest Worker ID, we get a nice evenish mix of the variations within each full run
+	// that also has variation throughout the day.
+	const variationIndex = ( currentHour + jestWorkerId ) % allVariations.length;
+	return allVariations[ variationIndex ];
```

Comment:
**comment:non-blocking (I think)** - I have some _minor_ reservations about this method. It does add an element of randomness, which is not ideal in tests and test selection, but is a really clever method to select a subset of tests without biasing coverage. 

I replicated what this is doing and simulated the output assuming `jestWorkerId < 2` and came up with this. Does this line up with what you were expecting?

```
----Hour 0
default
php-old
----Hour 1
php-old
php-new
----Hour 2
php-new
wp-beta
----Hour 3
wp-beta
wp-previous
----Hour 4
wp-previous
private
----Hour 5
private
ecomm-plan
----Hour 6
ecomm-plan
default
----Hour 7
default
php-old
----Hour 8
php-old
php-new
----Hour 9
php-new
wp-beta
----Hour 10
wp-beta
wp-previous
----Hour 11
wp-previous
private
----Hour 12
private
ecomm-plan
----Hour 13
ecomm-plan
default
----Hour 14
default
php-old
----Hour 15
php-old
php-new
----Hour 16
php-new
wp-beta
----Hour 17
wp-beta
wp-previous
----Hour 18
wp-previous
private
----Hour 19
private
ecomm-plan
----Hour 20
ecomm-plan
default
----Hour 21
default
php-old
----Hour 22
php-old
php-new
----Hour 23
php-new
wp-beta
```

My minor concern would be the void where no deploys are happening to Jetpack, which lands between 22:00 UTC - 07:00 UTC, roughly corresponding to when US East/Pacific finishes the day and Central Europe wakes up.

Reply:
For sure, those are great points, and I probably should have explained more of my reasoning!

> It does add an element of randomness, which is not ideal in tests and test selection, but is a really clever method to select a subset of tests without biasing coverage.

I definitely get this, and thought about it too... The reason I was okay with it was that we're only randomizing the _environment_, and the expectation is that these tests should be passing in all the environments. To me, it was a tradeoff then between missing intersecting bugs by hardcoding each specific test to an environment vs having some randomness, and I personally favor having the extra coverage! 🙂 It's also part of why I based it on hour + worker ID, it controls the randomness much more than true randomness.

> I replicated what this is doing and simulated the output assuming jestWorkerId < 2 and came up with this. My minor concern would be the void where no deploys are happening to Jetpack, which lands between 22:00 UTC - 07:00 UTC, roughly corresponding to when US East/Pacific finishes the day and Central Europe wakes up.

Two small notes on this!
1. We actually run this with 14 Jest workers, so it's two _per_ site, not two total! And I think this build should be run with Jest workers > 7 if possible. This build is far less likely to overwhelm an Atomic site compared to the full one because we're targeting different sites all at once (as opposed to the same).
2. Even if we did end up running this with a small number of workers, over the course of the normal Jetpack working hours, we would still get a mix of every variation, and that was the overall goal!